### PR TITLE
Issue 2695

### PR DESCRIPTION
**Pull request by dguhl**
Introduces localisation for API errors

Fixes issue #2695 

@JC5

### CODE REVIEW

File: app/Api/V1/Controllers/BudgetLimitController.php [hunk 1]
```diff
@@ -165,7 +165,7 @@ public function store(BudgetLimitRequest $request): JsonResponse
         $data   = $request->getAll();
         $budget = $this->repository->findNull($data['budget_id']);
         if (null === $budget) {
-            throw new FireflyException('Unknown budget.');
+            throw new FireflyException(trans('error_unknown_budget'));
```
Comment (JC5) [inline app/Api/V1/Controllers/BudgetLimitController.php]: Requires "api.".

File: app/Api/V1/Controllers/CategoryController.php [hunk 2]
```diff
@@ -163,7 +163,7 @@ public function store(CategoryRequest $request): JsonResponse
 
             return response()->json($manager->createData($resource)->toArray())->header('Content-Type', 'application/vnd.api+json');
         }
-        throw new FireflyException('Could not store new category.'); // @codeCoverageIgnore
+        throw new FireflyException(trans('error_store_new_category')); // @codeCoverageIgnore
```
Comment (JC5) [inline app/Api/V1/Controllers/CategoryController.php]: Requires "api.".

File: app/Api/V1/Controllers/ConfigurationController.php [hunk 3]
```diff
@@ -57,7 +57,7 @@ function ($request, $next) {
                 $admin = auth()->user();
 
                 if (!$this->repository->hasRole($admin, 'owner')) {
-                    throw new FireflyException('No access to method.'); // @codeCoverageIgnore
+                    throw new FireflyException(trans('error_no_access')); // @codeCoverageIgnore
```
Comment (JC5) [inline app/Api/V1/Controllers/ConfigurationController.php]: Requires "api.".

File: app/Api/V1/Controllers/LinkTypeController.php [hunk 4]
```diff
@@ -86,7 +86,7 @@ function ($request, $next) {
     public function delete(LinkType $linkType): JsonResponse
     {
         if (false === $linkType->editable) {
-            throw new FireflyException(sprintf('You cannot delete this link type (#%d, "%s")', $linkType->id, $linkType->name));
+            throw new FireflyException(trans('error_delete_link_type', [':id' => $linkType->id, ':name' => $linkType->name]));
```
Comment (JC5) [inline app/Api/V1/Controllers/LinkTypeController.php]: Requires "api.".

File: app/Api/V1/Controllers/TransactionLinkController.php [hunk 5]
```diff
@@ -165,7 +165,7 @@ public function store(TransactionLinkRequest $request): JsonResponse
         $inward  = $this->journalRepository->findNull($data['inward_id'] ?? 0);
         $outward = $this->journalRepository->findNull($data['outward_id'] ?? 0);
         if (null === $inward || null === $outward) {
-            throw new FireflyException('Source or destination is NULL.');
+            throw new FireflyException(trans('error_source_or_dest_null'));
```
Comment (JC5) [inline app/Api/V1/Controllers/TransactionLinkController.php]: Requires "api.".

File: app/Api/V1/Controllers/TransactionLinkController.php [hunk 6]
```diff
@@ -196,7 +196,7 @@ public function update(TransactionLinkRequest $request, TransactionJournalLink $
         $data['inward']  = $this->journalRepository->findNull($data['inward_id'] ?? 0);
         $data['outward'] = $this->journalRepository->findNull($data['outward_id'] ?? 0);
         if (null === $data['inward'] || null === $data['outward']) {
-            throw new FireflyException('Source or destination is NULL.');
+            throw new FireflyException(trans('api.error_source_or_dest_null'));
```
Comment (JC5) [inline app/Api/V1/Controllers/TransactionLinkController.php]: Requires "api.".

**Comment by dguhl** [comment]
Easily done.

File: app/Api/V1/Controllers/TransactionLinkController.php (hunk 6 above)
Comment (dguhl) [reply app/Api/V1/Controllers/TransactionLinkController.php]: It's already there. 🤔

Comment (JC5) [reply app/Api/V1/Controllers/TransactionLinkController.php]: Haha whoops

**Comment by JC5** [comment]
Nice work :+1:!

**Pull request merged**